core/src/main/java/de/evoila/cf/broker/service/impl/CatalogServiceImpl.java

```diff
@@ -144,4 +149,47 @@ public void filterActivePlans(Catalog catalog) {
 		});
 	}
 
+	public void controllAndManageMaintenanceInfo(Catalog catalog) {
+		if (catalog == null || catalog.getServices() == null) return;
+		for (ServiceDefinition definition : catalog.getServices()) {
+			if (definition.getPlans() != null) {
+				for (Plan plan : definition.getPlans()) {
+					MaintenanceInfo maintenanceInfo = plan.getMaintenanceInfo();
+					if (maintenanceInfo != null) {
+						String version = maintenanceInfo.getVersion();
+						if (StringUtils.isEmpty(version)) {
+							logger.error("Version field of maintenance_info for plan "
+									+ (plan.getId() == null ? "'ID NOT SET'" : plan.getId())
+									+ " is not set, but necessary if the object exists. Disabling it to prevent false configuration.");
+							plan.setMaintenanceInfo(null);
+						} else if (!checkIfVersionIsSemantic2(version)) {
+							logger.error("The configured version of the maintenance_info for plan "
+									+ (plan.getId() == null ? "'ID NOT SET'" : plan.getId())
+									+ "is not complying with required Semantic Versioning 2.0.0. Disabling it to prevent false configuration.");
+							plan.setMaintenanceInfo(null);
+							logger.info("######################");
+							logger.info("The version of the maintenance_info object for plan "+ (plan.getId() == null ? "'ID NOT SET'" : plan.getId()) + " is not complying to required Semantic Versioning 2.0.0");
```

Comment:
can we please preformat this string somewhere and only call logger.info one time.

Reply:
fixed